kafka/consumer.go

```diff
@@ -905,5 +905,5 @@ func (c *Consumer) handleRebalanceEvent(channel chan Event, rkev *C.rd_kafka_eve
 		c.events <- newError(cErr)
 	}
 
-	return nil
+	return ev
```

Comment:
I'm not sure it should return the event here, there is a code path that leads here even though a rebalanceCb is set, namely when the rebalanceCb did not make the assignment and the c.appReassigned was never set.
Since a rebalance callback is used poll should not return an event in this case either.
This would've been caught by the new tests if the rebalance callback did not call Assign(), but it does so for the AssignedPartitions case, but not RevokedPartitions. Unfortunately the test only triggers the assignment, not the revokal (I believe that's hidden in Close()).

So, I recommend to update the test to force a revoke after an assignment has been seen, easiest way is to call Unsubscribe(). That should fail the test with this current code.

Reply:
Hi @edenhill, thanks for reviewing. 

Close() is added at the test case, https://github.com/confluentinc/confluent-kafka-go/pull/645/files#diff-a9cef34b4369fffc694b2b116cb24394eb75fb0e5d67568a2114140b06c1652eR511, so revoke is triggered when the Close is called.
```
Local Test:
jliu@C02DM32YMD6T kafka % go test -run TestConsumerRebalanceEvents
Consumer closed
Consumer closed
COOPERATIVE rebalance: 4 new partition(s) assigned: [testConsumerRebalanceEvents[0]@unset testConsumerRebalanceEvents[1]@unset testConsumerRebalanceEvents[2]@unset testConsumerRebalanceEvents[3]@unset]
COOPERATIVE rebalance: 4 partition(s) revoked: [testConsumerRebalanceEvents[0]@unset testConsumerRebalanceEvents[1]@unset testConsumerRebalanceEvents[2]@unset testConsumerRebalanceEvents[3]@unset]
Consumer closed
EAGER rebalance: 4 new partition(s) assigned: [testConsumerRebalanceEvents[0]@unset testConsumerRebalanceEvents[1]@unset testConsumerRebalanceEvents[2]@unset testConsumerRebalanceEvents[3]@unset]
EAGER rebalance: 4 partition(s) revoked: [testConsumerRebalanceEvents[0]@unset testConsumerRebalanceEvents[1]@unset testConsumerRebalanceEvents[2]@unset testConsumerRebalanceEvents[3]@unset]
Consumer closed
PASS
ok  	_/Users/jliu/Documents/Code/jliunyu/confluent-kafka-go/kafka	0.198s
```

But if Call Unsubscribe() then call Close(), will see this error ```%3|1626911382.640|REBALANCE|rdkafka#consumer-3| [thrd:main]: Group "gotest": unexpected instruction to revoke current assignment and rebalance (terminating=0, assignment_lost=0, LEAVE_ON_UNASSIGN_DONE=0)```

Can you please help to conform the expected behavior? since both Unsubscribe() and Close() will trigger revoke, so Unsubscribe() will trigger the revoke if it's been called, otherwise close() will trigger it?